tf2_ros/src/create_timer_ros.cpp

```diff
@@ -36,15 +36,33 @@
 #include <functional>
 #include <mutex>
 #include <stdexcept>
+#include <memory>
 
 namespace tf2_ros
 {
 
 CreateTimerROS::CreateTimerROS(
   rclcpp::node_interfaces::NodeBaseInterface::SharedPtr node_base,
-  rclcpp::node_interfaces::NodeTimersInterface::SharedPtr node_timers)
-: node_base_(node_base), node_timers_(node_timers), next_timer_handle_index_(0)
+  rclcpp::node_interfaces::NodeTimersInterface::SharedPtr node_timers,
+  bool spin_thread)
+: node_base_(node_base), node_timers_(node_timers), next_timer_handle_index_(0),
+  spin_thread_(spin_thread)
 {
+  if (spin_thread_) {
+    callback_group_ = node_base->create_callback_group(
+      rclcpp::CallbackGroupType::MutuallyExclusive, false);
+    executor_ = std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
+    executor_->add_callback_group(callback_group_, node_base_);
+    dedicated_thread_ = std::make_unique<std::thread>([&]() {executor_->spin();});
```

Comment:
Question for maintainers: Would it make more sense to you to only have this thread running when in use (e.g. when we `create_timer` create this thread as well and when we reset / cancel it we destroy it then?)

Reply:
Yes, we we shouldn't start a thread if it's going to be unused. That will cause overhead and make debugging generally harder. Especially if there ends up potentially being several instances.